python_modules/dagster-ext/dagster_ext/_io/default.py

```diff
@@ -16,24 +17,45 @@
 )
 
 
-class ExtFileContextLoader(ExtContextLoader):
-    _path: Optional[str] = None
-
+class ExtDefaultContextLoader(ExtContextLoader):
     @contextmanager
     def load_context(self, params: ExtParams) -> Iterator[ExtContextData]:
-        path = assert_env_param_type(params, "path", str, self.__class__)
-        with open(path, "r") as f:
-            data = json.load(f)
-        yield data
-
+        if "path" in params:
+            path = assert_env_param_type(params, "path", str, self.__class__)
+            with open(path, "r") as f:
+                data = json.load(f)
+                yield data
+        elif "data" in params:
+            data = assert_env_param_type(params, "data", dict, self.__class__)
+            yield cast(ExtContextData, data)
+        else:
+            raise DagsterExtError(
+                f'Invalid params for {self.__class__.__name__}, expected key "path" or "data",'
+                f" received {params}",
+            )
 
-class ExtFileMessageWriter(ExtMessageWriter):
-    _path: Optional[str] = None
 
+class ExtDefaultMessageWriter(ExtMessageWriter):
     @contextmanager
-    def open(self, params: ExtParams) -> Iterator["ExtFileMessageChannel"]:
-        path = assert_env_param_type(params, "path", str, self.__class__)
-        yield ExtFileMessageChannel(path)
+    def open(self, params: ExtParams) -> Iterator[ExtMessageWriterChannel]:
+        if "path" in params:
+            path = assert_env_param_type(params, "path", str, self.__class__)
+            yield ExtFileMessageChannel(path)
+        elif "std" in params:
+            stream = assert_env_param_type(params, "std", str, self.__class__)
+            if stream == "err":
+                yield ExtStreamMessageChannel(sys.stderr)
+            elif stream == "out":
+                yield ExtStreamMessageChannel(sys.stdout)
+            else:
+                raise DagsterExtError(
+                    f'Invalid value for key "std", expected stderr or stdout but received {stream}'
+                )
+        else:
+            raise DagsterExtError(
+                f'Invalid params for {self.__class__.__name__}, expected key "path" or "std",'
+                f" received {params}"
```

Comment:
Can we encode the magic strings "path" "std" etc somewhere so it is clear what is going on and so we can match it  to code on the launching side?